server/portal/apps/projects/managers/base.py

```diff
@@ -0,0 +1,291 @@
+        )
+        return prj
+
+    def get_by_project_id(self, project_id):
+        """Get a single project.
+
+        :param str project_id: Project Id.
+        """
+        prj = Project(
+            self.user.agave_oauth.client,
+            project_id
+        )
+        if not prj.storage.uuid:
+            raise Exception("No project.")
+        return prj
+
+    def get_project(self, project_id=None, system_id=None):
+        """Get a single project.
+
+        This method will retrieve a project by, either, system_id or
+        project_id.
+
+        :param str project_id: Project Id (optional).
+        :param str system_id: System Id (optional).
+        """
+        if project_id:
+            prj = self.get_by_project_id(project_id)
+        elif system_id:
+            prj = self.get_by_system_id(system_id)
+        else:
+            raise Exception("Must use project_id or system_id.")
+        return prj
+
+    def create(self, title):
+        """Create a project.
+
+        This method will figure out the next consecutive number to assign to
+        the projectId and then call
+        :meth:`portal.apps.projects.models.base.Project.create`.
+        Meaning, it will create three objects:
+        1. Directory based on `settings.PORTAL_PROJECTS_ROOT_DIR`
+        2. Agave Metadata record with the project id and title.
+        3. Agave storage system pointing to the directory created.
+
+        :param str title: Project title
+        """
+        if not title:
+            raise Exception("No title.")
+
+        project_id = '{prefix}-{prj_id}'.format(
+            prefix=settings.PORTAL_PROJECTS_ID_PREFIX,
+            prj_id=ProjectId.next_id()
+        )
+        prj = Project.create(
+            self.user.agave_oauth.client,
+            title,
+            project_id,
+            self.user
+        )
+        prj.storage.update_role(
+            self.user.username,
+            'ADMIN'
+        )
+        METRICS.info(
+            'Project created: project_id=%s, project_title:%s, username=%s',
+            project_id, title, self.user.username
+        )
+
+        project_indexer.apply_async(args=[project_id])
+
+        return prj
+
+    def list(self, offset=0, limit=100):
+        """List projects."""
+        return [prj.storage for prj in Project.listing(
+            self.user.agave_oauth.client,
+            offset=offset,
+            limit=limit
+        )]
+
+    def add_member(self, project_id, member_type, username):
+        """Add member to a project.
+
+        When adding a member to a project we are giving
+        that user full access to everything within the project.
+
+        :param str project_id: Project Id.
+        :param str member_type: One of ['pi', 'co_pi', 'team_member']
+        :param str username: Username.
+        """
+        user = get_user_model().objects.get(username=username)
+
+        prj = self.get_project(project_id)
+        if member_type == 'team_member':
```

Comment:
While I realize this is a direct port of the backend, is this what we want? Do we want to implement any of the proposed changes to dynamic role names specified in metadata?

Reply:
I think direct port first.  then any possible changes in a PR later, right?

Reply:
Yeah let's do that - it will be easier for us to see and review changes as well